app/Jobs/SendNewRecordingNotifications.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Friend;
+use App\Jobs\Job;
+use App\Phone\Exceptions\BlacklistedPhoneNumberException;
+use App\Phone\TwilioClient;
+use App\PhoneNumber;
+use App\Recording;
+use App\User;
+use Illuminate\Log\Writer as Logger;
+
+class SendNewRecordingNotifications extends Job
+{
+
+    /**
+     * @var Recording
+     */
+    private $recording;
+
+    public function __construct(Recording $recording)
+    {
+        $this->recording = $recording;
+    }
+
+    public function handle(TwilioClient $twilio, Logger $logger)
+    {
+        $user = $this->recording->user;
+        $this->notifyOwner($twilio, $logger);
+        $this->notifyFriends($twilio, $logger, $user);
+    }
+
+    protected function notifyOwner(TwilioClient $twilio, Logger $logger)
+    {
+        $text = $this->getOwnerBody();
+        $logger->info($this->getOwnerNumber());
+
+        try {
+            $twilio->text($this->getOwnerNumber(), $text);
+        } catch (BlacklistedPhoneNumberException $e) {
+            $this->recording->phoneNumber->markBlacklisted();
+        }
+
+        $logger->info('Owner SMS sent: ' . $text);
+    }
+
+    protected function getOwnerNumber()
+    {
+        if ($number = $this->recording->phoneNumber) {
```

Comment:
Not sure of when we're gonna have a phoneNumber and when a recording. What's going on here? It's un-clear enough that I'm guessing we might want to document it in code.